### PR TITLE
Add ready Promise to Proc::Async

### PR DESCRIPTION
**Pull request by the-eater**
Adds the ready promise to Proc::Async, which keep's when the Proc is started, which may be useful in cases you don't control the start of the Proc but want to know when it starts

### CODE REVIEW

**Comment by lizmat** [comment]
Looks like a nice feature to me.

**Comment by moritz** [comment]
I like the idea, thanks!

I think the implementation can be improved. The code should take a vow on that promise (and keep it in a private attribute or so), so that other code cannot accidentally fulfill or break the promise, much like most code in the setting does that returns promises.

**Comment by the-eater** [comment]
@moritz is this what you mean?

File: src/core/Proc/Async.pm [hunk 1]
```diff
@@ -95,6 +97,10 @@ my class Proc::Async {
             $enc, :$translate-nl
     }
 
+    method ready(--> Promise) {
+      $!ready_promise;
```
Comment (jnthn) [inline src/core/Proc/Async.pm]: Should use 4-space indent here.

**Comment by the-eater** [comment]
Since "ready" is also the callback name used by MoarVM I thought it was a fitting name :)
I've already created a branch with tests https://github.com/the-eater/roast/tree/feature-proc-async-ready

Only docs still need to be done

File: src/core/Proc/Async.pm (hunk 1 above)
Comment (the-eater) [inline src/core/Proc/Async.pm]: Fixed :) thanks!

**Comment by moritz** [comment]
Thanks @the-eater!

**Pull request merged**